src/server/pkg/ppsutil/util.go

```diff
@@ -390,6 +390,10 @@ func IsTerminal(state pps.JobState) bool {
 
 // UpdateJobState performs the operations involved with a job state transition.
 func UpdateJobState(pipelines col.ReadWriteCollection, jobs col.ReadWriteCollection, jobPtr *pps.EtcdJobInfo, state pps.JobState, reason string) error {
+	if jobPtr.State == pps.JobState_JOB_FAILURE {
```

Comment:
I'm not sure if this is a good idea. It seems like it could mask bugs? I mean, I guess it is good to keep the job in a failed state if it was marked as failed before, but we would not know about the issue we found if this existed. The side effects from what you fixed (uploading a datums object, attempting egress, etc.) would go unnoticed without the job state indicating an issue.

Reply:
I don't feel strongly about keeping this. An alternative would be to return an error if the job is in FAILURE? That might actually have made the bug apparent earlier.

@jdoliner?

Reply:
Returning an error here seems like the right call.